### PR TITLE
Locksmith updated with delegated purchasing

### PR DESCRIPTION
**Pull request by akeem**
# Description

The code has been updated to support purchasing on a users
behalf. At the moment this is written with Ethereum based locks
in mind but shall be updated for others as well.

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [x] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by akeem** [comment]
While not standard, this change includes an update to package.json in addition to code. The cut over from the older version of the library create a chicken & egg situation

**Comment by akeem** [comment]
We also have a skipped test in this change for now, it will require a bit more wrangling and will be revisited in short order

File: locksmith/__tests__/controllers/purchaseController.test.ts [hunk 1]
```diff
@@ -57,6 +51,18 @@ describe('Purchase Controller', () => {
     })
   })
 
+  beforeEach(() => {
+    jest.mock('../../src/payment/paymentProcessor', () => {
```
Comment (cellog) [inline locksmith/__tests__/controllers/purchaseController.test.ts]: https://jestjs.io/docs/en/jest-object#jestgenmockfrommodulemodulename plus https://jestjs.io/docs/en/mock-function-api.html#mockfnmockimplementationoncefn may be a better way to do this that fixes the skipped test

File: locksmith/src/payment/paymentProcessor.ts [hunk 2]
```diff
@@ -90,21 +90,24 @@ export class PaymentProcessor {
     return Object.values(itemizedPrice).reduce((a, b) => a + b)
   }
 
-  initiatePurchase(
-    purchaser: ethereumAddress,
+  async initiatePurchase(
+    purchaser: ethereumAddress /** this is the managed user/buyer */,
```
Comment (cnasc) [inline locksmith/src/payment/paymentProcessor.ts]: I think `purchaser` would make more sense as `recipient`

**Pull request merged**